server/v2/cometbft/abci.go

```diff
@@ -230,14 +231,19 @@ func (c *Consensus[T]) PrepareProposal(ctx context.Context, req *abci.PreparePro
 		return nil, err
 	}
 
-	txs, err := c.app.BuildBlock(ctx, uint64(req.Height), uint32(cp.Block.MaxBytes))
+	txs, err := c.mempool.Get(ctx, int(cp.Block.MaxBytes))
+	if err != nil {
+		return nil, err
+	}
```

Comment:
if buildblock is linked to some logic like today how does this work? Right now this is a bit of blind picking of txs